### DISCUSSION TITLE
Want to change the texture in PIXI.SimplePlane and keep the deformation.

### DISCUSSION THREAD

**Discussion by cyocun** [pixijs/pixijs#8047] [Q&A]
Thanks for looking.

I am making a graphic that can be deformed with simplePlane by referring to this.
https://codepen.io/osublake/pen/8785174690eabaf5ba99f785b9d3b64e
https://codepen.io/osublake/pen/gLGYxx

Thanks to the wonderful codepen, I can now deform with simplePlane without any problems.

Now, I want to be able to replace the texture of the simplePlane, but when I replace the texture, the deformation is no longer applied.

I'm sure I'm wrong, but I'm very simply replacing the texture as follows.

```js
const plane = new PIXI.SImplePlane(texture,10,10)

// We use these vetices to transform it.
const vetices = plane.geometry.getBuffer('aVertexPosition');

. . .

// If we then replace the texture, the deformation will no longer be applied.
plane.texture = newTexture
```

Can you please tell me the correct procedure to replace the texture in SimplePlane?
I'm not familiar with the terminology, so I'm not sure if I should just destroy SimplePlane and create it again.

I am not familiar with the terminology, so please let me know if there are any mistakes in the language.
Also, I'm not familiar with geometry, so I'm sorry if I'm bothering you.
If you need more information, please let me know.

Thank you very much.

**Comment by domis86** [comment]
@cyocun can you try this tip? https://github.com/pixijs/pixijs/issues/7249#issuecomment-781034077

**Reply by cyocun** [reply]
@domis86

Yeah...
I did that after replacing the textures, but it didn't make any difference.
I also ran the following functions to see what I could find, but no luck.
```

    plane.textureUpdated();

    And then

    plane.updateTransform();
    plane.calculateUvs();
    plane.calculateVertices();
    plane.verticesBuffer.update();
    
    vetices.update();
```


_Sorry, I wasn't familiar with the discussion function, so I posted again._

**Reply by cyocun** [reply]
And.
I made a reproduction based on the following example from [#724](https://github.com/pixijs/pixijs/issues/7249#issuecomment-781034077) 
`Example: https://jsfiddle.net/bigtimebuddy/9ngx82Ld/`

Please check this -> https://codepen.io/cyocun/pen/vYegJpq

As shown in lines 160-163, replacing a texture with another one in the middle of the image does not deform it.
The sample is pixi v4, so please let me know if it's wrong at all.

Thanks for responding so quickly to my question!

**Reply by domis86** [reply]
i see you are using old PIxi version: "Pixi.js 4.5.1"
Can you try again using version 6 and SimplePlane (and then the `beach.textureUpdated();` method) ?

**Reply by cyocun** [reply]
Yes, I understood and was using version 4.5.1.

I just tried it again, and it works fine in both version 4.5.1 and version 6.2.0, 
as long as the textures are the same size! Thank you! 😆

Also, it worked fine without using plane.textureUpdated().
https://codepen.io/cyocun/pen/vYegJpq


However, it seems that if the texture size is different, it will not work even with version 6.2.0.

What I want is an answer for the same texture size, so this is the solution.

What should I do if the texture size is different?
https://codepen.io/cyocun/pen/vYegWBJ?editors=0010
